spmd/tensor/parallel/utils.py

```diff
@@ -44,6 +43,9 @@ def _prepare_input_validate(
     def wrapper(*args, **kwargs):  # pyre-ignore[2, 3]
         assert len(args) >= 1, "_prepare_input needs at least one arg."
         input = args[0]
+        if isinstance(input, list) or isinstance(input, tuple):
+            input = input[0]
+            args = (input, *args[1:])
```

Comment:
so if the first arg is a list of tuple, we only convert the first item in the list/tuple to DTensor? What if the first arg is sth like `(tensor, tensor, tensor)`?

Reply:
For now we can skip this case right? Or you feel this is important to handle?

Reply:
Addressed offline.